src/engine/execute.js

```diff
@@ -77,12 +77,10 @@ var execute = function (sequencer, thread) {
                 if (!edgeWasActivated) {
                     sequencer.retireThread(thread);
                 }
-            } else {
+            } else if (!resolvedValue) {
                 // Not an edge-activated hat: retire the thread
                 // if predicate was false.
-                if (!resolvedValue) {
-                    sequencer.retireThread(thread);
-                }
+                sequencer.retireThread(thread);
```

Comment:
Again on this one... I think the if inside the else might be a little easier to follow. This might always be the case so maybe that rule should be removed?